meshpy/four_c/dbc_monitor.py

```diff
@@ -88,44 +153,203 @@ def read_dbc_monitor_file(file_path):
+def add_point_neuman_condition_to_input_file(
+    input_file: InputFile,
+    nodes: list[int],
+    function_array: list[Function],
+    force: np.ndarray,
+    *,
+    n_dof: int = 3,
+):
     """
-    Convert the Dirichlet boundary condition monitor log to a Neumann
-    boundary condition input section.
+    Adds a Neumann boundary condition to the input file for the given node_ids
+    with the function_array and force values by creating a new geometry set.
 
     Args
     ----
     input_file: InputFile
-        The input file where the created Neumann boundary condition is added
-        to. The nodes referred to in the log file have to match with the ones
-        in the input section. It is advisable to only call this function once
-        all nodes have been added to the input file.
-    file_path: str
-        Path to the Dirichlet boundary condition log file.
-    step: int
-        Step values to be used. Default is -1, i.e. the last step.
-    function: Function, int
-        Function for the Neumann boundary condition.
+        InputFile where the boundary conditions are added to
+    nodes: [node_id]
+        list containing the ids of the nodes for the condition
+    function_array: [function]
+        list with functions
+    force: [np.ndarray]
+        values to scale the function array with
     n_dof: int
         Number of DOFs per node.
     """
 
-    nodes, _, force, _ = read_dbc_monitor_file(file_path)
+    # Add the function to the input file, if they are not previously added.
+    for function in function_array:
+        if function not in input_file.functions:
+            input_file.add(function)
 
-    # The forces are the negative reactions at the Dirichlet boundaries.
-    force *= -1.0
+    # check if the dimensions of force and functions match
+    if force.size != 3:
+        raise ValueError(
+            f"The forces vector must have dimensions [3x1] not [{force.size}x1]"
+        )
+
+    if len(function_array) != 3:
+        raise ValueError(f"The function array must have length 3 not {force.size}.")
 
-    # Create the BC condition for this set and add it to the input file.
+    # Create GeometrySet with nodes.
     mesh_nodes = [input_file.nodes[i_node] for i_node in nodes]
     geo = GeometrySet(mesh_nodes)
+
+    # Create the Boundary Condition.
     extra_dof_zero = " 0" * (n_dof - 3)
     bc = BoundaryCondition(
         geo,
         (
             "NUMDOF {n_dof} ONOFF 1 1 1{edz} VAL {data[0]} {data[1]} {data[2]}"
-            "{edz} FUNCT {{0}} {{0}} {{0}}{edz}"
-        ).format(n_dof=n_dof, data=force[step], edz=extra_dof_zero),
+            "{edz} FUNCT {{}} {{}} {{}}{edz}"
+        ).format(n_dof=n_dof, data=force, edz=extra_dof_zero),
         bc_type=mpy.bc.neumann,
-        format_replacement=[function],
+        format_replacement=function_array,
     )
     input_file.add(bc)
+
+
+def all_dbc_monitor_values_to_input(
```

Comment:
Maybe my last comment was not clear. I meant to keep the "original" `dbc_monitor_to_input` here, since it requires a much simpler interface compared to the more general version `all_dbc_monitor_values_to_input` (like you initially mentioned and I didn't think though 😬). So here we would simply have:
```python
def dbc_monitor_to_input(input_file, file_path, *, step=-1, function=1, **kwargs):
    """
    Convert the Dirichlet boundary condition monitor log to a Neumann
    boundary condition input section.

    Args
    ----
    input_file: InputFile
        The input file where the created Neumann boundary condition is added
        to. The nodes referred to in the log file have to match with the ones
        in the input section. It is advisable to only call this function once
        all nodes have been added to the input file.
    file_path: str
        Path to the Dirichlet boundary condition log file.
    step: int
        Step values to be used. Default is -1, i.e. the last step.
    function: Function
        Function for the Neumann boundary condition.
    """

    nodes, _, force, _ = read_dbc_monitor_file(file_path)

    # The forces are the negative reactions at the Dirichlet boundaries.
    force *= -1.0

    add_point_neumann_condition_to_input_file(
        input_file, nodes, [function] * 3, force[step], **kwargs
    )
```
This allows to keep the simple interface of the old functionality, while the actual code for adding the BC to the mesh is the same for the "old" and "new" function.

Reply:
Ok so I brought the function `dbc_monitor_to_input` back to life and provided 3 tests cases.
1) `test_four_c_simulation_dbc_monitor_to_input`  - tests the old functionality with the interface how it used to be. **Important:** Only one force value is extracted.  
2) `test_four_c_simulation_all_dbc_monitor_values_to_input` tests the same as test case 1) to ensure that the new function is capable of doing the same functionality.
3) `test_four_c_simulation_dirichlet_boundary_to_neumann_boundary_with_all_values` -  tests the new functionality as a function by selected force values is created and added.